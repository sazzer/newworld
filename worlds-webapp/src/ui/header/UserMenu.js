// @flow

import React from 'react';
import {connectStore} from "redux-box";
import type {UserModule} from '../../users';
import {module as usersModule} from '../../users';
import LoginMenu from './LoginMenu';
import LoggedInMenu from './LoggedInMenu';

/** The flow type representing the props for the User Menu */
type UserMenuProps = {
    hasCurrentUser: boolean
};

/**
 * React component to represent the area of the header bar that is clicked to log in
 */
export function UserMenu(props: UserMenuProps) {
    if (props.hasCurrentUser) {
        return <LoggedInMenu />;
    } else {
        return <LoginMenu />;
    }
}

/**
 * Wrapper around the User Menu that connects it to Redux
 */
export function ConnectedUserMenu({users}: {users: UserModule}) {
    return (
        <UserMenu hasCurrentUser={users.selectHasCurrentUser()} />
    )
}

export default connectStore({users: usersModule})(ConnectedUserMenu);
