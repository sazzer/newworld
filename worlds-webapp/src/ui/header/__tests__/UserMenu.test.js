// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {UserMenu} from '../UserMenu';

/** Create the component to test */
function setup({hasCurrentUser} = {hasCurrentUser: false}) {
    return {
        element: shallow(<UserMenu hasCurrentUser={hasCurrentUser} />)
    };
}

describe('UserMenu', () => {
    it('Renders correctly when no current user is present', () => {
        const { element } = setup({hasCurrentUser: false});

        expect(element).toMatchSnapshot();
    });

    it('Renders correctly when a current user is present', () => {
        const { element } = setup({hasCurrentUser: true});

        expect(element).toMatchSnapshot();
    });

});
