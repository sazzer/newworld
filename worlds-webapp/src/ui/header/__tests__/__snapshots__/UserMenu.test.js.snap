// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu Renders correctly when a current user is present 1`] = `<Connect(ConnectedLoggedInMenu) />`;

exports[`UserMenu Renders correctly when no current user is present 1`] = `<Connect(ConnectedLoginMenu) />`;
